### PR TITLE
ipaddr: Update docs

### LINKED ISSUE #50546: add more ipaddr netaddr examples into filter documentation
##### SUMMARY
integrate all these examples that work with the ipaddr filter
https://github.com/network-automation/ipaddr_filter

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html

##### ANSIBLE VERSION
N/A

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### ADDITIONAL INFORMATION
Will work on any laptop with ansible + netaddr installed `pip install netaddr` or you can literally test in sivel's browser tool https://ansible.sivel.net/test/

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Added docs for

* next_nth_usable
* next_usable
* range_usable
* last_usable

Fixes: #50546

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst

### CODE REVIEW

**Comment by IPvSean** [comment]
looks great!
great job

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 1]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
 
 .. _6to4: https://en.wikipedia.org/wiki/6to4
 
+Getting additional information from network address
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can obtain next usable IP address from the given network address using ``next_usable`` ::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
To find the next usable IP address in a range, use ``next_usable`` ::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 2]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
 
 .. _6to4: https://en.wikipedia.org/wiki/6to4
 
+Getting additional information from network address
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:410]: This section would be stronger with an over-arching description of what these filters do. I'm guessing a bit on the wording, but something like this:

```suggestion

To find usable IP addresses within an IP range, try these ``ipaddr`` filters:

```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 3]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
...
+You can obtain next usable IP address from the given network address using ``next_usable`` ::
+
+    # {{ '192.168.122.1/24' | ipaddr('next_usable') }}
+    192.168.122.2
+
+Similarly, you can obtain the last usable IP address from the given network address using ``last_usable``::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
To find the last usable IP address from a range, use ``last_usable``::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 4]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
...
+
+    # {{ '192.168.122.1/24' | ipaddr('last_usable') }}
+    192.168.122.254
+
+You can use ``range_usable``, if you want to find the available range of IP addresses from the given network address, like::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
To find the available range of IP addresses from the given network address, use ``range_usable``::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 5]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
...
+
+    # {{ '192.168.122.1/24' | ipaddr('range_usable') }}
+    192.168.122.1-192.168.122.254
+
+You can use ``next_nth_usable`` filter to find out next nth usable IP address from the given network address::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
To find the next nth usable IP address within a range, use ``next_nth_usable``::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 6]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
...
+
+    # {{ '192.168.122.1/24' | next_nth_usable(2) }}
+    192.168.122.3
+
+In above example, ``next_nth_usable`` returns 2nd usable IP address for the given IP range.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
In this example, ``next_nth_usable`` returns the 2nd usable IP address for the given IP range.
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 7]
```diff
@@ -405,6 +405,32 @@ be automatically converted to a router address (with a ``::1/48`` host address):
 
 .. _6to4: https://en.wikipedia.org/wiki/6to4
 
+Getting additional information from network address
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Finding IP addresses within a range
```

**Review by acozine — COMMENTED** [review]
This is a great addition to the filter documentation, thanks @Akasurde.

I've added some suggestions to make the examples easier to find and parse.

**Pull request merged**